lib/blemulator/internal/bridge/platform_to_dart_bridge.dart

```diff
@@ -11,14 +11,15 @@ class PlatformToDartBridge {
   }
 
   Future<dynamic> _handleCall(MethodCall call) async {
-    String transactionId = call.arguments[SimulationArgumentName.transactionId];
-    Future<dynamic> platformCallResultFuture = _dispatchPlatformCall(call);
-
-    if (transactionId != null) {
+    if (call.method != DartMethodName.cancelTransaction &&
+        call.arguments.containsKey(SimulationArgumentName.transactionId)) {
+      String transactionId =
+          call.arguments[SimulationArgumentName.transactionId];
+      _cancelTransactionIfExists(transactionId);
       return _createAndSaveCancelablePlatformCall(
```

Comment:
How about changing this method to just accept MethodCall, and pulling the argument out inside? _dispatchPlatformCall could be hidden from here as well then.